### DISCUSSION TITLE
Sort by column clicking on raven UI

### DISCUSSION THREAD

**Discussion by Jooseba** [ravendb/ravendb#16850] [Ideas]
Hi,

RavenDB UI is one of the best I have seen on database management, just one small thing would make it perfect. When documents are queried and the columns of fields are shown, there could be an action to sort result by column just by clicking column and it would use just some regular alphanumeric sorting or date sorting if it identifies that the field is date etc.

![image](https://github.com/ravendb/ravendb/assets/44804211/682eb089-3e3d-4443-9060-0a38926590d3)

Here is an example photo what I mean.

**Comment by ayende** [comment]
That is a small feature on the surface, but has huge implications.
Consider the following scenario:

![image](https://github.com/ravendb/ravendb/assets/116915/0fbbbc76-5bac-4e0b-9471-abb40cc44ccf)


As you can see, there are about 59.5M (!) records in this collection.
If I would try to sort it by `created_at`, what will happen?

Assuming that you don't have an index on `created_at` already, we have the following choices:

* Run a normal query, which RavenDB will translate to an auto index. That means that a single click in the Studio can generate an index.
* Don't run a query, iterate over the values and sort them directly.

To give some context, this is a 208GB database.

In either option, you'll need to read all of that to memory and do a stupendous amount of work.

Making that happen as a result of a single click (or worse, a "miss-click") is horrible

**Reply by Jooseba** [reply]
But wouldn't that be same operation that querying by manually with order by?

![image](https://github.com/ravendb/ravendb/assets/44804211/5f5000b9-bd61-424d-8d43-daedf170660b)

And for miss-click there could be an confirmation or an options which enables the feature.

**Reply by ayende** [reply]
Yes, that is exactly what will happen.
But querying in an active process, and we expect that either:

* The user is aware of RavenDB behavior in this case
* The dataset is small enough it doesn't matter
* The admin disabled auto index generation if they aren't desired

The UI element in question is not subject to the same constraints. As a user, I would expect that to be something cheap, not "spike all resource usage for hours" in some cases.

Adding an alert would simply make the process more awkward, not a good idea
